lib/controller_patches.rb

```diff
@@ -6,6 +6,16 @@
 #
 Rails.configuration.to_prepare do
   HelpController.class_eval do
+    before_action :set_history
+
     def house_rules; end    
+
+    private
+
+    def set_history
+      @history ||= HelpPageHistory.new(
+        lookup_context.find_template("#{controller_path}/#{action_name}")
+      )
+    end
```

Comment:
Can you explain what this will do? I know enough rails that it's going to run before every action method on the controller ( get, update, destroy, list...) and do a query. 

What's the query? can it be cached? Is the query unique to the user? It looks like it's the same date for everyone, but it's going to run on every view of the page.